File: backend/src/main/java/com/example/ubernet/model/Driver.java
package com.example.ubernet.model;

import com.example.ubernet.model.enums.AuthProvider;
import com.example.ubernet.model.enums.UserRole;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;

@EqualsAndHashCode(callSuper = true)
@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class Driver extends User {
    @OneToOne
    private Car car;
    @ManyToOne
    private DriverDailyActivity driverDailyActivity;
    private boolean requestedProfileChanges;
    public Driver(String email, String name, String surname, Car car) {
        super(email, name, surname);
        this.car = car;
    }

    public Driver(String email) {
        super(email);
    }
}

File: frontend/src/app/model/Driver.ts
export class Driver {
  email!: string;
  name!: string;
  surname!: string;
  city!: string;
  phoneNumber!: string;
  role!: string;
  isWorking: boolean;

  constructor(name: string, surname: string, email: string, phoneNumber: string, city: string, isWorking: boolean) {
    this.name = name;
    this.surname = surname;
    this.email = email;
    this.city = city;
    this.phoneNumber = phoneNumber;
    this.isWorking = isWorking;
  }
}
